tests/unit/test_charm.py

```diff
@@ -1719,7 +1719,7 @@ def test_scope_obj(harness):
 
 
 @patch_network_get(private_address="1.1.1.1")
-def test_get_secret(harness):
+def test_get_secret_from_databag(harness):
     with patch("charm.PostgresqlOperatorCharm._on_leader_elected"):
```

Comment:
Renaming is great! I'd add a docstring/comment highlighting that this test is for shared functionality between Juju2 and Juju3 as we have backwards compatibility support on reading from the databag.